src/main/java/qna/domain/Question.java

```diff
@@ -75,19 +60,29 @@ public boolean isOwner(User loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
+    public void delete(User loginUser, List<DeleteHistory> deleteHistories) throws CannotDeleteException {
```

Comment:
가능하면 인자로 전달 받은 값을 변경하기 보다는 값을 만들어 리턴하는게 좋습니다.
지금은 `deleteHistories` 가 List 자료구조여서 크게 와닿지 않을 수 있지만, 일종의 setter를 사용하는 것과 같다고 생각해요.
그리고 인자로 전달 받은 값에 문제가 생겼을 때 디버깅도 어렵구요. (지금도 여러 메서드에서 사용되고 있죠)

아래와 같은 구조로 변경해보면 어떨까요?
```suggestion
    public List<DeleteHistory> delete(User loginUser) throws CannotDeleteException {
        List<DeleteHistory> deleteHistories = new ArrayList<>();
        // ...
        return deleteHistories;
    }    
```

Reply:
네 ~ 안그래도 이 부분이 고민되더라구요 리팩토링 진행하겠습니다 감사합니다 :)